mantidproject/mantid
# Changed LoadSampleEnvironment to use angles instead of rotation matrix

~~**DO NOT MERGE UNTIL #25643 IS MERGED**~~
**Description of work.**
Changed `LoadSampleEnvironment` to take an angle of rotation on the x y and z axis instead of a matrix of rotation, the angles are applied in the same order.

**To test:**
update WSFile and StlFile so they point at the your local test data, then run the following script
```
WSFile = '/home/sjenkins/Work/Build-1/ExternalData/Testing/Data/SystemTest/WISH/input/18_1/WISH00040503.raw'
StlFile = "/home/sjenkins/Work/Build-1/ExternalData/Testing/Data/UnitTest/cube.stl"

Load(Filename=WSFile, OutputWorkspace='ws',SpectrumList="100")
ConvertUnits(InputWorkspace="ws",Target="Wavelength", OutputWorkspace="ws1")
ConvertUnits(InputWorkspace="ws",Target="Wavelength", OutputWorkspace="ws2")
LoadSampleShape(InputWorkspace="ws1",Filename=StlFile,Scale="mm",OutputWorkspace="ws1")
SetSampleMaterial(InputWorkspace="ws1",ChemicalFormula="V",SampleNumberDensity=0.1)
LoadSampleEnvironment(InputWorkspace="ws1",Filename=StlFile,Scale="cm",OutputWorkspace="ws1",SetMaterial=True,ChemicalFormula="V",SampleNumberDensity=0.1,XDegrees=90)
MonteCarloAbsorption(InputWorkspace="ws1",OutputWorkspace="rotatedAbsorb",NumberOfWavelengthPoints=50)
LoadSampleShape(InputWorkspace="ws2",Filename=StlFile,Scale="mm",OutputWorkspace="ws2")
SetSampleMaterial(InputWorkspace="ws2",ChemicalFormula="V",SampleNumberDensity=0.1)
LoadSampleEnvironment(InputWorkspace="ws2",Filename=StlFile,Scale="cm",OutputWorkspace="ws2",SetMaterial=True,ChemicalFormula="V",SampleNumberDensity=0.1)
MonteCarloAbsorption(InputWorkspace="ws2",OutputWorkspace="absorb",NumberOfWavelengthPoints=50)

```
then plot `absorb` and `rotatedAbsorb`,`rotatedAbsord` should be larger

Fixes #25650 
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
